src/Handler/CommandNameExtractor/NamedCommandExtractor.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace League\Tactician\Handler\CommandNameExtractor;
+
+/**
+ * Extract the name from a NamedCommand
+ */
+class NamedCommandExtractor implements CommandNameExtractor
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getNameForCommand($command)
+    {
+        if ($command instanceof NamedCommand) {
+            return $command->getName();
+        }
+
+        // fallback to class name
+        return get_class($command);
```

Comment:
The most consistent would be CanNotDetermineCommandNameException, let's just run with that.

I'd still be against against the fallback here. Most of the time, folks will choose one strategy and stick with it throughout their app. I think the setup makes it easy to do custom strategies and that's totally alright (and supported!). I'd just rather see folks create their own strategies than making this part more complex.